src/CoreWCF.Http/src/CoreWCF/Channels/HttpRequestContext.cs

```diff
@@ -375,10 +382,34 @@ public override HttpOutput GetHttpOutput(Message message)
                 return HttpOutput.CreateHttpOutput(_aspNetContext, HttpTransportSettings, message, HttpMethod);
             }
 
-            protected override SecurityMessageProperty OnProcessAuthentication()
+            protected override async Task<SecurityMessageProperty> OnProcessAuthenticationAsync()
             {
-                // TODO: Wire up authentication for ASP.Net Core
-                //return Listener.ProcessAuthentication(listenerContext);
+                if (HttpTransportSettings.IsAuthenticationRequired)
+                {
+                    AuthenticateResult authenticateResult = _aspNetContext.Features.Get<AuthenticateResult>();
+                    if (authenticateResult == null)
+                    {
+                        // No authentication occurred
+                        return null;
+                    }
+
+                    if (authenticateResult.None)
+                    {
+                        throw new Exception("The request requires authentication, but the user was not authenticated.");
+                    }
+                    else
+                    {
+                        ServiceSecurityContext securityContext = await CreateSecurityContextAsync(authenticateResult);
+                        SecurityMessageProperty securityMessageProperty = new SecurityMessageProperty
+                        {
+                            ServiceSecurityContext = securityContext
+                        };
+                        return securityMessageProperty;
+                    }
+                }
+
+                // Authentication is required but we were unable to authenticate.
+                // Should we throw instead?
```

Comment:
You only get here if HttpTransportSettings.IsAuthenticationRequired returns false, so it makes sense to not have a SecurityMessageProperty.